parachain/pallets/control/src/lib.rs

```diff
@@ -198,19 +202,18 @@ pub mod pallet {
 		pub fn create_channel(origin: OriginFor<T>) -> DispatchResult {
 			let location: MultiLocation = T::ChannelOrigin::ensure_origin(origin)?;
 
-			let (agent_id, some_para_id, _) = Self::convert_location(location)?;
+			let (agent_id, para_id, location) = Self::convert_location(location)?;
 
 			ensure!(Agents::<T>::contains_key(agent_id), Error::<T>::AgentNotExist);
 
-			let para_id = some_para_id.ok_or(Error::<T>::LocationToParaIdConversionFailed)?;
-
 			ensure!(!Channels::<T>::contains_key(para_id), Error::<T>::ChannelAlreadyCreated);
 
 			Channels::<T>::insert(para_id, ());
 
 			let message = Message {
-				origin: T::OwnParaId::get(),
+				origin: para_id,
```

Comment:
This is a fix for other control operations except `create_agent`, the origin should be the original Parachain, not bridgeHub.